samples/snake/src/main/kotlin/example/utils/PackingUtils.kt

```diff
@@ -0,0 +1,17 @@
+@file:Suppress("NOTHING_TO_INLINE")
+
+package example.utils
+
+import androidx.collection.IntIntPair
+
+inline fun IntIntPair.asLong(): Long {
+	return packInts(first, second)
+}
+
+inline fun packInts(val1: Int, val2: Int): Long {
+	return (val1.toLong() shl 32) or (val2.toLong() and 0xFFFFFFFF)
+}
+
+inline val Long.first: Int get() = (this shr 32).toInt()
+
+inline val Long.second: Int get() = (this and 0xFFFFFFFF).toInt()
```

Comment:
The idea of these properties is to get the first and second values from `Long` (the internal representation of `IntIntPair`), although yes, it looks like a common utility thing. I renamed it to match my idea.
In general, these extensions are needed, since in our version `Long` inside `IntIntPair` is still internal.